src/rendergraph/opengl/backend/basegeometrynode.cpp

```diff
@@ -0,0 +1,115 @@
+#include "backend/basegeometrynode.h"
+
+#include <QOpenGLTexture>
+#include <stdexcept>
+
+#include "backend/shadercache.h"
+#include "rendergraph/engine.h"
+#include "rendergraph/geometrynode.h"
+#include "rendergraph/texture.h"
+
+using namespace rendergraph;
+
+namespace {
+GLenum toGlDrawingMode(DrawingMode mode) {
+    switch (mode) {
+    case DrawingMode::Triangles:
+        return GL_TRIANGLES;
+    case DrawingMode::TriangleStrip:
+        return GL_TRIANGLE_STRIP;
+    default:
+        throw std::runtime_error("not implemented");
```

Comment:
What's our take with exception? AFAIK, there are pretty rare in the code base. IMHO, it wouldn't be a bad idea to start using them, but we should catch them. Otherwise, we could use the `DEBUG_ASSERT` pattern.
@Swiftb0y wdyt?

Reply:
I think the consensus was that using exceptions currently will result in brittle code (because you never know what throws and when) so the consensus was to avoid exceptions currently and instead use `std::expected`. In this case a dummy value or a `std::optional` could work.